Parametrize # of longest tests in print_test_stats

samestep opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45941 Parametrize # of longest tests in print_test_stats**

This adds CLI options to the `test/print_test_stats.py` script for specifying how many of the longest tests should be printed. It also makes the following incidental changes:
- The script now has a `--help` option to describe its usage.
- The number of longest tests being shown is now displayed as a number, rather than in words.
- The median time is now displayed with the label `median_time` instead of `mean_time`, is calculated using `statistics.median` instead of raw indexing and bit shifting, and is displayed even when there are only two tests in a class.

Differential Revision: [D24154491](https://our.internmc.facebook.com/intern/diff/D24154491)

samestep committed:
Parametrize # of longest tests in print_test_stats

[ghstack-poisoned]

walterddr commented on test/print_test_stats.py:
```diff
@@ -42,21 +43,19 @@ def append(self, test_case):
         self.errored_count += 1 if test_case.errored else 0
 
-    def print_report(self):
+    def print_report(self, num_longest):
```
nit: for backward compatibility since `print_report` is not `_print_report`
```suggestion
    def print_report(self, num_longest=3):
```

walterddr commented on test/print_test_stats.py:
```diff
@@ -143,13 +169,12 @@ def send_report(reports):
         longest_tests.extend(test_suite.test_cases)
-        if len(longest_tests) > 10:
-            longest_tests = sorted(longest_tests, key=lambda x: x.time)[-10:]
+    longest_tests = sorted(longest_tests, key=lambda x: x.time)[-args.longest_of_run:]
```
good catch!

walterddr reviewed (approved):
thanks for the changes @samestep looks good!

samestep committed:
Use default num_longest=3 in test/print_test_stats.py

Co-authored-by: Rong Rong <rongr@fb.com>

samestep committed:
Update on "Parametrize # of longest tests in print_test_stats"


This adds CLI options to the `test/print_test_stats.py` script for specifying how many of the longest tests should be printed. It also makes the following incidental changes:
- The script now has a `--help` option to describe its usage.
- The number of longest tests being shown is now displayed as a number, rather than in words.
- The median time is now displayed with the label `median_time` instead of `mean_time`, is calculated using `statistics.median` instead of raw indexing and bit shifting, and is displayed even when there are only two tests in a class.

Differential Revision: [D24154491](https://our.internmc.facebook.com/intern/diff/D24154491)

[ghstack-poisoned]
